sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpRedirectOptions.java

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+import com.generic.core.http.Response;
+import com.generic.core.http.policy.RequestRedirectCondition;
+import com.generic.core.models.HeaderName;
+import com.generic.core.util.ClientLogger;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.function.Predicate;
+
+/**
+ * Options to configure the redirect policy's behavior.
+ */
+public final class HttpRedirectOptions {
+    private static final ClientLogger LOGGER = new ClientLogger(HttpRedirectOptions.class);
+    private final int maxAttempts;
+    private final Set<HttpMethod> allowedRedirectHttpMethods;
+    private final HeaderName locationHeader;
+    private Predicate<RequestRedirectCondition> shouldRedirectCondition;
+
+    /**
+     * Creates an instance of {@link HttpRedirectOptions} with values for {@code maxAttempts}
+     * defaulting to 3, if not specified, and the default set of allowed HTTP methods {@link HttpMethod#GET}
+     * and {@link HttpMethod#HEAD} and the default header name "Location" to locate the redirect url in the response
+     * headers.
+     *
+     * @param maxAttempts The maximum number of redirect attempts to be made.
+     * @param allowedRedirectHttpMethods The set of HTTP methods that are allowed to be redirected.
+     * @param locationHeader The header name containing the redirect URL.
+     * @throws IllegalArgumentException if {@code maxAttempts} is less than 0.
+     */
+    public HttpRedirectOptions(int maxAttempts, HeaderName locationHeader, Set<HttpMethod> allowedRedirectHttpMethods) {
```

Comment:
Should think about using `EnumSet` rather than `Set` as `HttpMethod` is an enum (it's more performant)